app/assets/javascripts/hydra-editor/manage_repeating_fields.js

```diff
@@ -3,139 +3,171 @@
+            $(this.element).find(this.listClass).children('li').last().find('.form-control').focus();
+        },
+
+        addToList: function( event ) {
             event.preventDefault();
-            var $activeField = $(event.target).parents(this.fieldWrapperClass)
+            var $listing = $(event.target).closest('.multi_value').find(this.listClass)
+            $activeField = $listing.children('li').last()
 
             if (this.inputIsEmpty($activeField)) {
                 this.displayEmptyWarning();
             } else {
-                var $listing = $(this.listClass, this.element);
                 this.clearEmptyWarning();
                 $listing.append(this._newField($activeField));
             }
-          },
-
-          inputIsEmpty: function($activeField) {
-              return $activeField.children('input.multi-text-field').val() === '';
-          },
-
-          _newField: function ($activeField) {
-              var $newField = this.createNewField($activeField);
-              // _changeControlsToRemove must come after createNewField
-              // or the new field will not have an add button
-              this._changeControlsToRemove($activeField);
-              return $newField;
-          },
-
-          createNewField: function($activeField) {
-              $newField = $activeField.clone();
-              $newChildren = $newField.children('input');
-              $newChildren.val('').removeProp('required');
-              $newChildren.first().focus();
-              this.element.trigger("managed_field:add", $newChildren.first());
-              return $newField
-          },
-
-          _changeControlsToRemove: function($activeField) {
-              var $removeControl = this.remover.clone();
-              $activeFieldControls = $activeField.children('.field-controls');
-              $('.add', $activeFieldControls).remove();
-              $activeFieldControls.prepend($removeControl);
-          },
-
-          clearEmptyWarning: function() {
-              $listing = $(this.listClass, this.element),
-              $listing.children(this.warningClass).remove();
-          },
-
-          displayEmptyWarning: function () {
-              $listing = $(this.listClass, this.element)
-              var $warningMessage  = $("<div class=\'message has-warning\'>cannot add new empty field</div>");
-              $listing.children(this.warningClass).remove();
-              $listing.append($warningMessage);
-          },
-
-          removeFromList: function( event ) {
-            event.preventDefault();
 
-            var field = $(event.target).parents(this.fieldWrapperClass)
-            field.remove();
+            this._manageFocus()
+        },
+
+        inputIsEmpty: function($activeField) {
+            return $activeField.children('input.multi-text-field').val() === '';
+        },
+
+        _newField: function ($activeField) {
+            var $newField = this.createNewField($activeField);
```

Comment:
Is there any reason to assign to a variable, or can you just return the result of the function call?